### PR TITLE
Update to new fuel-vm version

### PR DESCRIPTION
**Pull request by Dentosal**
Includes breaking changes from https://github.com/FuelLabs/fuel-vm/pull/307. Requires https://github.com/FuelLabs/fuel-vm/pull/314 as well.

### CODE REVIEW

**Comment by Dentosal** [comment]
I totally missed that PR, it indeed seems to have lots of duplicate changes. I'm going to wait until it's merged and rebase this on top of it after.

**Review by Voxelot — APPROVED** [review]

File: crates/fuel-core/src/database/balances.rs
Comment (bvrooman) [inline crates/fuel-core/src/database/balances.rs:94]: 👍

**Review by bvrooman — APPROVED** [review]

**Pull request merged**
